packages/mui-material/src/InputBase/InputBase.js

```diff
@@ -439,8 +439,7 @@ const InputBase = React.forwardRef(function InputBase(inProps, ref) {
     if (inputRef.current && event.currentTarget === event.target) {
       inputRef.current.focus();
     }
-
-    if (onClick) {
+    if (onClick && !fcs.disabled) {
```

Comment:
Why? This doesn't make sense to me. A click event should bubble up, regarded of the disabled nature.

Reply:
🤔 This looked ok to me as a plain `<input>` doesn't fire `onClick` when it's disabled:  https://codesandbox.io/s/async-https-n9x4y3-onclick-disabled-n9x4y3

Reply:
But there can be startAdornment, endAdornment, and renderSuffix. IMHO, we shouldn't try to make InputBase behave like an input, it's not, it's Root + input.

Reply:
Oh, I didn't realize this also changed `<TextField onClick>` behavior https://github.com/mui/mui-x/issues/9386#issuecomment-1643642932. For this, one, no way, props should be spread on the root element, unless they absolutely should be on a lower element, which is not the case here.